stored_requests/caches/memory/cache_test.go

```diff
@@ -54,36 +49,44 @@ func TestRaceUnboundedConcurrency(t *testing.T) {
 		TTL:              -1,
 	})
 
+	doRaceTest(t, cache)
+}
+
+func doRaceTest(t *testing.T, cache stored_requests.Cache) {
 	done := make(chan struct{})
-	go writeLots(cache, done, 100)
-	go readLots(cache, done, 100)
-	go invalidateLots(cache, done, 100)
+	sets := [][]int{rand.Perm(100), rand.Perm(100), rand.Perm(100)}
```

Comment:
I would create separate random sets for each goroutine, otherwise not much benefit beyond going in order, as they are all still going in the same order, just not the 1, 2, 3, 4 order.